tests/phpunit/CRM/Event/Form/ParticipantTest.php

```diff
@@ -48,6 +48,23 @@ public function testSubmit(): void {
     $this->callAPISuccessGetSingle('Participant', ['id' => $form->getParticipantID()]);
   }
 
+  public function testSubmitWithExistingParticipant(): void {
+    $form = $this->getForm([], [
+      'register_date' => date('Ymd'),
+      'status_id' => 1,
+      'role_id' => [CRM_Core_PseudoConstant::getKey('CRM_Event_BAO_Participant', 'role_id', 'Attendee')],
+    ])->postProcess();
+    $this->assertEquals($this->getEventID(), $form->getEventID());
+    $this->callAPISuccessGetSingle('Participant', ['id' => $form->getParticipantID()]);
+    $participantID = $form->getParticipantID();
+
+    $form = new CRM_Event_Form_Participant();
+    $form->_action = CRM_Core_Action::ADD;
+    $form->_id = $participantID;
```

Comment:
Hmm - add would not have a participant ID - I think you are perhaps trying to trick the form here since you are setting internal properties directly - I took a look & was able to write it using the form harness